src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -535,16 +532,25 @@ private void Resize(int newSize, bool forceNewHashCodes)
                 }
             }
 
+            int k = 0;
             for (int i = 0; i < count; i++)
```

Comment:
Does the loop need to be to `_entries.Count` now?

Reply:
It may be profitable to cache `_entries` in a local. The JIT should be able to eliminate the bounds check that way:

```
Entry[] oldEntries = _entries;
for (int i = 0; i < oldEntries.Count; i++)
```